site/src/job_queue.rs

```diff
@@ -36,6 +41,93 @@ async fn create_benchmark_request_master_commits(
     Ok(())
 }
 
+/// Parses strings in the following formats extracting the Date & release tag
+/// `static.rust-lang.org/dist/2016-05-24/channel-rust-1.9.0.toml`
+/// `static.rust-lang.org/dist/2016-05-31/channel-rust-nightly.toml`
+/// `static.rust-lang.org/dist/2016-06-01/channel-rust-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89.0-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89.0-beta.2.toml`
+fn parse_release_string(url: &str) -> anyhow::Result<Option<(String, DateTime<Utc>)>> {
```

Comment:
Hmm, we already have `parse_published_artifact_tag` and `missing_published_artifacts`. But it's true that we didn't parse the release dates before, which was a shame, as they are missing in the DB, and it was using the DB index with artifact SHAs for checking what was missing. So I suppose that this temporary duplication is fine.